### ISSUE TITLE
weekly activities 2.0

### ISSUE THREAD

**Issue by OliverLSanz** [goaliath-app/goaliath#10]
Weekly activities will appear on the "Today" screen after they are chosen "to do today". A special activity called "select weekly activities" will appear and take you to the selection screen.

- [x] implement the UI for the new SelectWeekliesScreen
- [x] add a weekliesSelected boolean field to the log slice
- [x] make the special activity "select weekly activities" appear in TodayScreen
- [x] modify the log generation so that logs for weekly activities are just added to the current day when they are selected in the SelectWeekliesScreen
- [x] connect SelectWeekliesScreen to Redux
- [x] make logs of weekly activities appear in TodayScreen
- [x] remove WeeklyScreen

**Comment by OliverLSanz** [comment]
By now the feature is working in the weekly-activities-v2 branch. We should do the following improvements before considering it finished.

- [ ] completed weekly activities should not be eligible to be added to today. They may still appear to make the user proud of their progress!
- [ ] when an activity is selected, 1 day should be subtracted from its "days remaining" and the result should appear in a different color.
- [ ] weekly activities should be clearly differentiated in the todayScreen, maybe by writing "weekly" at the start of the description.
- [ ] We should use the Checkbox component present in TodayScreen for the check-boxes in the SelectWeeklyActivitiesScreen. The current ones have a small touchable surface and are not centered.
- [ ] the top weekly visualization has yet to be added, as designed in Figma

**Comment by OliverLSanz** [comment]
Problem detected! We had to make entries of weekly activities not to be created by the updateLog function. To achieve this I modified the dueToday function to return false for any weekly activity. This caused the updateLog function to not create those logs, but also meant that It the function will always archive any weekly entry present. To fix this I changed the updateLog function so it entirely ignores weekly activities. BUT HERE IS THE PROBLEM: the updateLog function don't archive logs of weekly activities when the activity or its goal is deleted, archived or disabled. We have to fix that. So, the problem is that
- Entries of weekly activities don't get archived when the activity or goal is disabled, deleted or archived

Also, I noticed more odd behavior:
- disabled weekly activities still appear in the ChooseWeekliesScreen
- weekly activities of disabled goals also appear
- even deleted (archived) activities appear
- and they also appear if their goal has been deleted (archived)
- if there is not any entry for today, the "choose weeklies" activities won't appear, even if there are weekly activities to choose.
